### PR TITLE
PEOPLE-105 rewrite remaining feature specs

### PR DESCRIPTION
**Pull request by mczuchnowski**
**Changes:**
- total refactor of users index page spec
- refactor of user profile spec
- tiny changes in factories

**Jira:**
https://netguru.atlassian.net/browse/PEOPLE-105
https://netguru.atlassian.net/browse/PEOPLE-101

This is the final batch for these tickets.

### CODE REVIEW

File: spec/features/users_spec.rb [hunk 1]
```diff
@@ -28,12 +28,45 @@
...
+    end
+  end
+
+  context 'potential booked project' do
+    let!(:potential_project) { create(:project, :potential) }
+    let!(:membership) { create(:membership, :booked, project: potential_project, user: developer) }
+
+    before { users_page.load }
+
+    it "shows potential sign" do
```
Comment (netguru-hound) [inline spec/features/users_spec.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: spec/features/teams_spec.rb [hunk 2]
```diff
@@ -67,28 +67,26 @@
...
     let(:new_team_name) { 'Relatively OK team' }
-    let(:success_msg) { "Team #{new_team_name} changed successfully" }
-    let(:error_msg) { 'New name not provided' }
 
     it 'shows edit form' do
       expect(teams_page).to have_edit_team_modal
     end
 
-    # TODO: find out why failing on circle
+    # TODO find out why this won't work on CircleCI
```
Comment (netguru-hound) [inline spec/features/teams_spec.rb]: Annotation keywords like `TODO` should be all upper case, followed by a colon, and a space, then a note describing the problem.

File: spec/features/users_spec.rb [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
+  before do
+    log_in_as developer
+    users_page.load
   end
 
-  it 'shows user role' do
-    names = developer.primary_roles.pluck(:name).join(', ')
-    expect(page).to have_content(names)
+  it 'shows user roles' do
+    developer.primary_roles.each { |role| expect(page).to have_content(role.name) }
```
Comment (micjez) [inline spec/features/users_spec.rb]: you could use matcher regex here instead of loop

**Comment by micjez** [comment]
perfect, approved

**Pull request merged**